library/Icinga/Crypt/AesCrypt.php

```diff
@@ -76,6 +79,28 @@ public function setMethod($method)
         return $this;
     }
 
+    /**
+     * Get supported method
+     *
+     * @return string
+     */
+    public function getSupportedMethod()
+    {
+        $availableMethods = openssl_get_cipher_methods();
+        foreach (self::METHODS as $method) {
+            if ($this->isRequiredTag() && ! $this->isAESupported()) {
+                $this->method = null;
+                continue;
+            }
```

Comment:
Why is this part of the loop? It does nothing with `$method`, even resets `$this->method` which the entire function body doesn't touch.

Reply:
This checks if the method set by default supports the php version, if not, it simply jumps to the next index 1. So we can override the default method if php does not support it.

Reply:
I know what it does. But it has no connection to the current iteration in any way. If the first cipher isn't the GCM one, it ignores it which would be wrong.

If you want to invalidate the default method, don't set one. Isn't `getSupportedMethod()` designed to perform this job? If PHP doesn't allow us to use GCM, it shouldn't report it.